### PR TITLE
fix(ingest/snowflake) - Fixing snowflake url with default region

### PR DESCRIPTION
**Pull request by treff7es**
## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_utils.py [hunk 1]
```diff
@@ -82,7 +82,7 @@ def create_snowsight_base_url(
         if privatelink:
             url = f"https://app.{account_locator}.{cloud_region_id}.privatelink.snowflakecomputing.com/"
         elif cloud == SNOWFLAKE_DEFAULT_CLOUD:
-            url = f"https://app.snowflake.com/{cloud_region_id}/{account_locator}/"
+            url = f"https://app.snowflake.com/{cloud_region_id}.{SNOWFLAKE_DEFAULT_CLOUD}/{account_locator}/"
```
Comment (mayurinehate) [inline metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_utils.py:85]: It looks like - there is some peculiar behavior in snowflake when it comes to "aws". Please see - https://docs.snowflake.com/en/user-guide/admin-account-identifier#non-vps-account-locator-formats-by-cloud-platform-and-region

`.aws` is not added in account identifier for some cloud regions in aws - namely - `us-west-2, us-east-1, eu-west-1, eu-central-1, ap-southeast-1, ap-southeast-2` . We would need to not append `.aws` for these regions.

Comment (treff7es) [reply metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_utils.py]: thanks, I updated based on this

File: metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_utils.py [hunk 2]
```diff
@@ -79,12 +80,25 @@ def create_snowsight_base_url(
         cloud: str,
         privatelink: bool = False,
     ) -> Optional[str]:
+        if cloud:
+            url_cloud_provider_suffix = f".{cloud}"
+
+        if cloud == SnowflakeCloudProvider.AWS:
+            if cloud_region_id in [
```
Comment (mayurinehate) [inline metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_utils.py:87]: Can we please extract these regions list as a constant and mark this constant with the snowflake link -> from where we figured out these ?

**Review by mayurinehate — APPROVED** [review]
lgtm

**Review by hsheth2 — APPROVED** [review]

**Comment by hsheth2** [comment]
One lint issue, and also the `tests/integration/snowflake/test_snowflake.py::test_snowflake_basic` needs to be updated

```
Skipped 1 files
src/datahub/ingestion/source/snowflake/snowflake_utils.py:10:1: F401 'datahub.ingestion.source.snowflake.constants.SNOWFLAKE_DEFAULT_CLOUD' imported but unused
1     F401 'datahub.ingestion.source.snowflake.constants.SNOWFLAKE_DEFAULT_CLOUD' imported but unused
```

**Comment by hsheth2** [comment]
The airflow test failures are unrelated and will be fixed by #9434

**Pull request merged**